app/components/MintBTC.tsx

```diff
@@ -79,78 +139,119 @@ interface MintNBTCForm {
 }
 
 export function MintBTC() {
-	const { balance: walletBalance } = useXverseWallet();
-	const { isWalletConnected } = useContext(WalletContext);
+	const { toast } = useToast();
+	const [txId, setTxId] = useState<string | undefined>(undefined);
+	const { connectWallet } = useXverseConnect();
+	const { balance: walletBalance, currentAddress } = useXverseWallet();
+	const { isWalletConnected, suiAddr } = useContext(WalletContext);
 	const isBitCoinWalletConnected = isWalletConnected(Wallets.Xverse);
-	const balance = parseBTC(walletBalance ?? "0");
+
 	const mintNBTCForm = useForm<MintNBTCForm>({
 		mode: "all",
 		reValidateMode: "onChange",
 		defaultValues: {
 			numberOfBTC: "",
-			suiAddress: "",
+			suiAddress: suiAddr || "",
 		},
 	});
 
 	const { handleSubmit, watch, setValue } = mintNBTCForm;
-	const numberOfBTC = watch("numberOfBTC");
+	const SuiAddress = watch("suiAddress");
 
-	const youReceive = parseBTC(numberOfBTC || "0") - nBTCMintFeeInSatoshi;
+	useEffect(() => setValue("suiAddress", suiAddr || ""), [setValue, suiAddr]);
 
 	return (
 		<FormProvider {...mintNBTCForm}>
 			<form
-				onSubmit={handleSubmit((formData) => {
-					// TODO: handle the nbtc form data
-					console.log("Depositing with Sui Address:", formData);
+				onSubmit={handleSubmit(async ({ numberOfBTC, suiAddress }) => {
```

Comment:
move to a local function. Too much nesting.

Reply:
Done